### DISCUSSION TITLE
Ulf Hammarqvist: WebCodecs don't have "packetization"/"containerization"?

### DISCUSSION THREAD

**Discussion by kgzoom** [w3c/webcodecs#796] [Q&A] [answered]
I wanted to know what Ulf Hammarqvist meant when he was saying that:

"WebCodecs is very nice, but there's something missing in the sense that we don't have packetization, or containerization or whatever the correct term would be."

Does this have anything to do with frames of audio or video? Maybe someone could explain in digital audio terms?

Source:

https://www.w3.org/2021/03/media-production-workshop/talks/ulf-hammarqvist-audio-latency.html

**Comment by padenot** [comment, marked as answer]
There is no muxing / demuxing capabilities in the Web Codecs API, it only deals with encoding and decoding images and audio, not containerization.

e.g. this spec deals with encoding the image sequence in h264 and the pcm audio in AAC, but it doesn't provide anything to take those h264 and AAC packets into an MP4 container.
